modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionFactory.java

```diff
@@ -180,14 +180,24 @@ public EndpointDefinition createDefinition(OMElement elem) {
             OMElement action = timeout.getFirstChildWithName(
                     new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "responseAction"));
             if (action != null && action.getText() != null) {
-                String actionString = action.getText();
-                if ("discard".equalsIgnoreCase(actionString.trim())) {
-                    definition.setTimeoutAction(SynapseConstants.DISCARD);
-                } else if ("fault".equalsIgnoreCase(actionString.trim())) {
-                    definition.setTimeoutAction(SynapseConstants.DISCARD_AND_FAULT);
+                String actionString = action.getText().trim();
+                if (isExpression(actionString)) {
+                    try {
+                        String expressionStr = actionString.substring(1, actionString.length() - 1);
+                        SynapseXPath expressionXPath = new SynapseXPath(expressionStr);
```

Comment:
this is restricted to xpath expressions. we need jsonPath and synapse-expression support also

Reply:
Added synapse-expression instead of xpath expressions.